EasyFinance.Application/Features/ExpenseService/ExpenseService.cs

```
@@ -19,76 +23,129 @@ public ExpenseService(IUnitOfWork unitOfWork)
+
+            var expenses = category.Expenses;
+
+            return AppResponse<IEnumerable<ExpenseResponseDTO>>.Success(expenses.ToDTO());
         }
 
-        public async Task<Expense> GetByIdAsync(Guid expenseId)
+        public async Task<AppResponse<ExpenseResponseDTO>> GetByIdAsync(Guid expenseId)
         {
-            return await this.unitOfWork.ExpenseRepository.Trackable()
+            var expense = await unitOfWork.ExpenseRepository.Trackable()
                 .Include(e => e.Items.OrderBy(item => item.Date))
-                    .ThenInclude(e => e.CreatedBy)
+                .ThenInclude(e => e.CreatedBy)
                 .Include(e => e.Attachments)
                 .Include(e => e.CreatedBy)
                 .FirstOrDefaultAsync(p => p.Id == expenseId);
+
+            if (expense == null)
+                return AppResponse<ExpenseResponseDTO>.Error(code: nameof(expense), description: string.Format(ValidationMessages.ExpenseNotFound, nameof(expense)));
+
+            return AppResponse<ExpenseResponseDTO>.Success(expense.ToDTO());
         }
 
-        public async Task<Expense> CreateAsync(User user, Guid categoryId, Expense expense)
+        public async Task<AppResponse<ExpenseResponseDTO>> CreateAsync(User user, Guid categoryId, Expense expense)
         {
             if (expense == default)
-                throw new ArgumentNullException(nameof(expense), string.Format(ValidationMessages.PropertyCantBeNullOrEmpty, nameof(expense)));
+                AppResponse<ExpenseResponseDTO>.Error(code: nameof(expense), description: string.Format(ValidationMessages.PropertyCantBeNullOrEmpty, nameof(expense)));
 
             if (user == default)
-                throw new ArgumentNullException(nameof(user), string.Format(ValidationMessages.PropertyCantBeNullOrEmpty, nameof(user)));
+                AppResponse<ExpenseResponseDTO>.Error(code: nameof(user), description: string.Format(ValidationMessages.PropertyCantBeNullOrEmpty, nameof(user)));
```

```suggestion
                return AppResponse<ExpenseResponseDTO>.Error(code: nameof(user), description: string.Format(ValidationMessages.PropertyCantBeNullOrEmpty, nameof(user)));
```
I suggest return here or the code will just run and ignore the error.